### PR TITLE
Add table oci_logging_log. Closes #60

### LINKED ISSUE #60: Add table oci_logging_log
**Is your feature request related to a problem? Please describe.**
We need to know if for all subnets in VCN is enabled with flow log or not. Part of CIS 3.14 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/oci_logging_log []

PRETEST: tests/oci_logging_log
Running terraform
oci_logging_log_group.named_test_resource: Creating...
oci_logging_log_group.named_test_resource: Creation complete after 2s [id=ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaad3zf4yga6f3upqnazozhdgirbks4u6cqvk3ubpzs3ifa]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci logging log create --display-name steampipetest3746 --log-group-id ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaad3zf4yga6f3upqnazozhdgirbks4u6cqvk3ubpzs3ifa --log-type CUSTOM"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.named_test_resource (local-exec): {
null_resource.named_test_resource (local-exec):   "opc-work-request-id": "ocid1.logworkrequest.oc1.ap-mumbai-1.aaaaaaaanlmzlovylikploxlfqerndjsxgqc4lqexakpm554avspbw5bhahq"
null_resource.named_test_resource (local-exec): }
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci logging log list --log-group-id ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaad3zf4yga6f3upqnazozhdgirbks4u6cqvk3ubpzs3ifa --output json > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/oci_logging_log/terraform/pretest/output.json"]
null_resource.named_test_resource: Still creating... [20s elapsed]
null_resource.named_test_resource: Still creating... [30s elapsed]
null_resource.named_test_resource: Creation complete after 30s [id=505349684079358716]
data.local_file.input: Reading...
data.local_file.input: Read complete after 0s [id=e9445b6b5b60f7fde0e0f19091bbae80d9d9ce77]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

display_name = "steampipetest3746"
log_group_id = "ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaad3zf4yga6f3upqnazozhdgirbks4u6cqvk3ubpzs3ifa"
region = "ap-mumbai-1"
resource_id = "ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaa7uqh364c54b3ydxmrhwcrt4yt3tqlmtq47hcajo4rzaq"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

TEST: tests/oci_logging_log
Running terraform

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Running SQL query: test-get-query.sql

[
  {
    "id": "ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaa7uqh364c54b3ydxmrhwcrt4yt3tqlmtq47hcajo4rzaq",
    "lifecycle_state": "ACTIVE",
    "name": "steampipetest3746",
    "title": "steampipetest3746"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "id": "ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaa7uqh364c54b3ydxmrhwcrt4yt3tqlmtq47hcajo4rzaq",
    "name": "steampipetest3746"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest3746"
  }
]
✔ PASSED

POSTTEST: tests/oci_logging_log
Running terraform
null_resource.destroy_test_resource: Creating...
null_resource.destroy_test_resource: Provisioning with 'local-exec'...
null_resource.destroy_test_resource (local-exec): Executing: ["/bin/sh" "-c" "    oci logging log delete --log-group-id ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaad3zf4yga6f3upqnazozhdgirbks4u6cqvk3ubpzs3ifa --log-id ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaa7uqh364c54b3ydxmrhwcrt4yt3tqlmtq47hcajo4rzaq --force\n"]
null_resource.destroy_test_resource: Still creating... [10s elapsed]
null_resource.destroy_test_resource (local-exec): {
null_resource.destroy_test_resource (local-exec):   "opc-work-request-id": "ocid1.logworkrequest.oc1.ap-mumbai-1.aaaaaaaawdvrwraodw2fcyuhzty7qb6mdlkqbd3taqc4rmje7hv5wkgixxta"
null_resource.destroy_test_resource (local-exec): }
null_resource.destroy_test_resource: Creation complete after 16s [id=5208857939081847391]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

TEARDOWN: tests/oci_logging_log



SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  id,
  log_group_id,
  name,
  lifecycle_state,
  time_created
from
  oci_logging_log;
+----------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+--------------------
| id                                                                                     | log_group_id                                                                                | name               
+----------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+--------------------
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaah5ebcd2tjy4bs4p7mb5f5oku5istfzanwwxiaqeofxea | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaai6m6c2ateqigqx6lt7t2qo6sm5rqd7qhx4yj3ym5jhaa | bucket_20210308_192
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaak4ejfp7ekp7fen4zbynfihpe566b2xo6tdowuojm3iaq | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaai6m6c2ateqigqx6lt7t2qo6sm5rqd7qhx4yj3ym5jhaa | testinh_new_region_
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaagtzncalw4j5c3hxnw5anlgj7x24chxitxomovvm4de5a | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaai6m6c2ateqigqx6lt7t2qo6sm5rqd7qhx4yj3ym5jhaa | test1              
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaaml5s3cxpifrwguqmu3kgswdxqib4q4swek5mvyclzvsa | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaai6m6c2ateqigqx6lt7t2qo6sm5rqd7qhx4yj3ym5jhaa | test               
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaafahri5i5ieoti7ap4b7mcrsom2vwz5yfxdsmjerleskq | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaap53g63tilticge55dwhnd6zylxtpyygoyqxu3klm6qaa | steampipetest5549  
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaabpdwrn2jrdwggvdywxgyfop47cbdawhxgth2szd4psmq | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaap53g63tilticge55dwhnd6zylxtpyygoyqxu3klm6qaa | steampipetest4986  
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaayertpho3hcvpmpmvbhi2jpjqraqjbp65i3az2pa3xjoq | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaaq2fo4o2reqnxm2nromzz3vrst4yq7fpbb3clrsvy6zka | test1              
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaaa2ljvqnk57xn7mktlkokbvihsqes6lgxqcqmeqxeoupq | ocid1.loggroup.oc1.ap-mumbai-1.amaaaaaa6igdexaaq2fo4o2reqnxm2nromzz3vrst4yq7fpbb3clrsvy6zka | teret              
| ocid1.log.oc1.iad.amaaaaaa6igdexaae6ebs2yrlmvpkqc7brgut75w5hzauswulxacjc7cykcq         | ocid1.loggroup.oc1.iad.amaaaaaa6igdexaa2okabqoeyjyeqw2c5xzytiwisko5m3mijvf4jn43g7dq         | testqw             
+----------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+--------------------
> select
  id,
  name,
  lifecycle_state as state,
  time_created
from
  oci_logging_log
where
  lifecycle_state = 'INACTIVE';
+----------------------------------------------------------------------------------------+------+----------+---------------------+
| id                                                                                     | name | state    | time_created        |
+----------------------------------------------------------------------------------------+------+----------+---------------------+
| ocid1.log.oc1.ap-mumbai-1.amaaaaaa6igdexaaml5s3cxpifrwguqmu3kgswdxqib4q4swek5mvyclzvsa | test | INACTIVE | 2021-04-20 05:41:30 |
+----------------------------------------------------------------------------------------+------+----------+---------------------+

select
  configuration -> 'source' ->> 'resource' as subnet_id,
  configuration -> 'source' ->> 'service' as service,
  lifecycle_state
from
  oci_logging_log
where
  configuration -> 'source' ->> 'service' = 'flowlogs'
  and lifecycle_state = 'ACTIVE';
+-------------------------------------------------------------------------------------------+----------+-----------------+
| subnet_id                                                                                 | service  | lifecycle_state |
+-------------------------------------------------------------------------------------------+----------+-----------------+
| ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaa6iyqkvgennyalwd5wt2t76zxwinlhg6rwrnbcxltextjnjdttdbq | flowlogs | ACTIVE          |
+-------------------------------------------------------------------------------------------+----------+-----------------+

```
</details>

### CODE REVIEW

File: docs/tables/oci_logging_log.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+The Oracle Cloud Infrastructure Logging service is a highly scalable and fully managed single pane of glass for all the logs in your tenancy. Logging provides access to logs from Oracle Cloud Infrastructure resources. These logs include critical diagnostic information that describes how resources are performing and being accessed.
+
+## Examples
+
+### Basic info
+
+```sql
+select
+  id,
+  name,
```
Comment (khushboo9024) [inline docs/tables/oci_logging_log.md:12]: Please add `log_group_id` column also in this query.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Integration test logs are missing and also this branch has `log_group` changes , please rebase with main and do the required changes.

**Comment by cbruno10** [comment]
Closing draft PR, @bigdatasourav please re-open when ready for review.

**Comment by bigdatasourav** [comment]
test issue resolved now using CLI

File: oci-test/tests/oci_logging_log/variables.tf [hunk 2]
```diff
@@ -29,13 +34,26 @@ provider "oci" {
...
 }
 
-resource "oci_logging_log" "named_test_resource" {
-  log_group_id = oci_logging_log_group.named_test_resource.id
-  display_name = var.resource_name
-  log_type = "Custom log"
+resource "null_resource" "named_test_resource" {
+  depends_on = [oci_logging_log_group.named_test_resource]
+  provisioner "local-exec" {
+    command = "oci logging log create --display-name ${var.resource_name} --log-group-id ${oci_logging_log_group.named_test_resource.id} --log-type ${var.log_type}"
```
Comment (khushboo9024) [inline oci-test/tests/oci_logging_log/variables.tf:47]: Is resource getting deleted as we are creating resource from CLI ?

File: oci/table_oci_logging_log.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	return &plugin.Table{
+		Name:        "oci_logging_log",
+		Description: "OCI Logging Log",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"id", "log_group_id"}),
+			Hydrate:    getLoggingLog,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listLoggingLogGroups,
+			Hydrate:       listLoggingLog,
```
Comment (khushboo9024) [inline oci/table_oci_logging_log.go:27]:
```suggestion
			Hydrate:       listLoggingLogs,
```

File: oci/table_oci_logging_log.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+				Hydrate:     getTenantId,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listLoggingLog(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline oci/table_oci_logging_log.go:139]:
```suggestion
func listLoggingLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: oci/table_oci_logging_log.go [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+	// Restrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+	logGroupId := d.KeyColumnQuals["log_group_id"].GetStringValue()
+
+	// handle empty internet gateway id in get call
```
Comment (khushboo9024) [inline oci/table_oci_logging_log.go:195]:
```suggestion
	// handle empty log and log group id in get call
```

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Minor changes required.

File: oci-test/tests/oci_logging_log/variables.tf (hunk 2 above)
Comment (bigdatasourav) [reply oci-test/tests/oci_logging_log/variables.tf:47]: updated with posttest tf.

File: oci-test/tests/oci_logging_log/pretest-variables.tf [hunk 6]
```diff
@@ -0,0 +1,79 @@
+variable "resource_name" {
+  type        = string
+  default     = "steampipetest20200125"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "tenancy_ocid" {
+  type        = string
+  default     = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
```
Comment (khushboo9024) [inline oci-test/tests/oci_logging_log/pretest-variables.tf:9]: Remove this.

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**